Explorer/Assets/DCL/AvatarRendering/AvatarShape/Systems/AvatarLoaderSystem.cs

```diff
@@ -53,6 +60,37 @@ private void UpdateAvatar(ref PBAvatarShape pbAvatarShape, ref AvatarShapeCompon
             pbAvatarShape.IsDirty = false;
         }
 
+        // TODO: remove PBAvatarShape as middleware, instead use Profile directly
+        [Query]
+        private void UpdateAvatarByProfile(in Entity entity, ref Profile profile, ref PBAvatarShape pbAvatarShape, ref PartitionComponent partition)
+        {
+            Color avatarSkinColor = profile.Avatar.SkinColor;
+            Color avatarHairColor = profile.Avatar.HairColor;
+
+            World.Set(entity, new PBAvatarShape
+            {
+                Id = profile.UserId,
+                BodyShape = profile.Avatar.BodyShape,
+                Wearables = { profile.Avatar.SharedWearables },
+                Name = profile.Name,
+                SkinColor = new Color3
+                {
+                    R = avatarSkinColor.r,
+                    B = avatarSkinColor.b,
+                    G = avatarSkinColor.g,
+                },
+                HairColor = new Color3
+                {
+                    R = avatarHairColor.r,
+                    B = avatarHairColor.b,
+                    G = avatarHairColor.g,
+                },
+                IsDirty = true,
+            });
+
+            World.Remove<Profile>(entity);
```

Comment:
It could be useful to keep the profile in the player's entity. However the avatar should be updated just once, when the profile changes. WDYT?

Reply:
It makes sense to keep `Profile` on the same entity as in the future you will probably need to detect if the profile has changed and to update the avatar

Reply:
I've reworked how Avatar Shape Component is created from Profile to make the flow more prod-ready